Lecture7/Testing_Task1/Helper/OnlinerUtils/OnlinerNavigation.java

```diff
@@ -0,0 +1,87 @@
+
+    private OnlinerNavigation() {
+    }
+
+    public static void goToOnliner(WebDriver driver) {
+        driver.get(onlinerURI);
+        System.out.println("Onliner was opened.");
+    }
+
+    public static void loginToOnliner(WebDriver driver) {
+        driver.findElement(By.cssSelector("div.auth-bar__item--text")).click();
+        driver.findElement(By.cssSelector("input.auth-input[type=text]"))
+              .sendKeys(onlinerLogin);
+        driver.findElement(By.cssSelector("input.auth-input[type=password]"))
+              .sendKeys(onlinerPassword);
+        driver.findElement(By.cssSelector("button.auth-button")).submit();
+        System.out.println("Login to Onliner was performed");
+    }
+
+    public static void openOnlinerFullCatalog(Wait<WebDriver> waiter) {
+        waiter.until(ExpectedConditions.visibilityOfElementLocated(By.className("b-main-navigation__link"))).click();
+        System.out.println("Onliner full catalog was opened");
+    }
+
+    public static void openOnlinerRandomCatalogChapter(WebDriver driver) {
+        List<WebElement> classifiersLinks = driver.findElements(By.className("catalog-navigation-classifier__item "));
+        int classifiersLinksIndex = (int) (Math.random() * classifiersLinks.size());
+        classifiersLinks.get(classifiersLinksIndex).click();
+        System.out.println("Classifier: " + classifiersLinks.get(classifiersLinksIndex).getText());
+        WebElement selectedCatalogBlock = driver.findElement(By.xpath("//div[@class=\"catalog-navigation-list__category\" and @style=\"display: block;\"]"));
+        List<WebElement> categoriesLinks = selectedCatalogBlock.findElements(By.className("catalog-navigation-list__aside-item"));
+        int categoriesLinksIndex = (int) (Math.random() * categoriesLinks.size());
+        categoriesLinks.get(categoriesLinksIndex).click();
+        System.out.println("Category: " + categoriesLinks.get(categoriesLinksIndex)
+                                                         .findElement(By.className("catalog-navigation-list__aside-title"))
+                                                         .getText());
+        WebElement activeCategory = driver.findElement(By.className("catalog-navigation-list__aside-item_active"));
+        List<WebElement> activeCategoryChapters = activeCategory.findElements(By.className("catalog-navigation-list__dropdown-item"));
+        int activeCategoryChaptersIndex = (int) (Math.random() * activeCategoryChapters.size());
+        System.out.println("Chapter: " + activeCategoryChapters.get(activeCategoryChaptersIndex)
+                                                               .findElement(By.className("catalog-navigation-list__dropdown-title"))
+                                                               .getText());
+        activeCategoryChapters.get(activeCategoryChaptersIndex).click();
+    }
+
+    public static void openOnlinerRandomProductWithOffers(WebDriver driver, Wait<WebDriver> waiter) {
```

Comment:
why not use like variable for all class Helper?

Reply:
fixed